### PR TITLE
ISPN-10137 Make annotated classes, fields, and methods accessible to the annotation processor

### PR DESCRIPTION
**Pull request by danberindei**
The component annotation processor required changing some of the rules for annotated components:

 * Annotated classes, fields, and methods must be package-private
 * `@Scope` must be present on each class or interface with injection or lifecycle annotations
 * `@MBean` must be present on each class or interface with managed attributes or operations
 * `@Scope` and `@MBean` are inherited, but only from classes
 * If `@SurvivesRestarts` is present on a class/interface, all subclasses must have it
 * Anonymous classes are not supported

Partly because of the rule changes, #6948 is pretty big, so I've extracted (most of) the code changes required in this PR. Despite its size, it should be trivial to review.

### CODE REVIEW

File: core/src/test/java/org/infinispan/manager/CacheManagerTest.java [hunk 1]
```diff
@@ -616,4 +611,23 @@ public void stop() {
...
+                                   CompletableFuture<Void> managerStopResumed) {
+         this.managerStopBlocked = managerStopBlocked;
+         this.managerStopResumed = managerStopResumed;
+      }
+
+      @Override
+      public void stop() {
+         log.tracef("Stopping global component registry");
+         managerStopBlocked.complete(null);
+         managerStopResumed.join();
```
Comment (karesti) [inline core/src/test/java/org/infinispan/manager/CacheManagerTest.java]: one is one complete and the other join ?

File: core/src/test/java/org/infinispan/persistence/ActivationDuringEvictTest.java [hunk 2]
```diff
@@ -106,7 +106,7 @@ private void assertPassivated(DataContainer dc, CacheLoader cl, String key, Stri
       assertEquals(expected, se.getValue());
    }
 
-   private static class SlowDownInterceptor extends CommandInterceptor implements Cloneable{
+   static class SlowDownInterceptor extends CommandInterceptor implements Cloneable{
```
Comment (karesti) [inline core/src/test/java/org/infinispan/persistence/ActivationDuringEvictTest.java]: space Cloneable

File: core/src/test/java/org/infinispan/manager/CacheManagerTest.java (hunk 1 above)
Comment (danberindei) [reply core/src/test/java/org/infinispan/manager/CacheManagerTest.java]: Yes, it's moved from https://github.com/infinispan/infinispan/pull/6978/files/e2737cb7482462a0e67fbdc1d561697163d8be67#diff-1ed883ada0e6bc47a12ece1101e8b891L280

File: core/src/test/java/org/infinispan/persistence/ActivationDuringEvictTest.java (hunk 2 above)
Comment (danberindei) [reply core/src/test/java/org/infinispan/persistence/ActivationDuringEvictTest.java]: Fixed

**Comment by karesti** [comment]
@danberindei if CI is fine, I will merge

**Comment by danberindei** [comment]
@karesti it's fine, the failures are unrelated

**Comment by tristantarrant** [comment]
Merged

**Pull request merged**